server/pages/index.js

```diff
@@ -388,15 +405,23 @@ function setUpLoggedInRoute( req, res, next ) {
 					}
 
 					console.error( 'API Error: ' + errorMessage );
-
-					next( error );
 				}
 
-				return;
+				return error;
 			} );
-	} else {
-		next();
+
+		setupRequests.push( userPromise );
 	}
+
+	Promise.all( setupRequests ).then( results => {
+		const rejectedWithErrors = results.filter( r => null != r && r.error );
```

Comment:
We avoid type-agnostic comparisons in Calypso, so this should be `r !== null` (we don;t care so much about Yoda case, thanks to type-aware comparisons).

Or you lodash's `get`: `r => get( r, [ 'error' ] )`

Better yet, I think you can even drop the `get` and combine this line with the logic two lines below by using lodash's [`some`](https://lodash.com/docs/4.17.10#some):

```js
if ( some( setupRequests, 'error' ) ) {
	return next( rejectedWithErrors );
}
```

Reply:
(I might be mistaken here, resolving the promise early might work just fine, if I'm reading `Promise.all()` docs corectly.)

Reply:
As the simplification of af8d712, there is no tricky `rejectedWithErrors` anymore.